### ISSUE TITLE
New rule: lifecycle param name

### ISSUE THREAD

**Issue by golopot** [yannickcr/eslint-plugin-react#2289]
The rule enforces that lifecycle parameter names exactly matches the well known names. For example, a `prevProps` should not be named as `props`, because people might confuse it with `this.props`.

This are considered **warnings**:
```js
class Foo extends Component {
  componentDidUpdate(props) {}
//                   ~~~~~ Error: should be `prevProps`.
  shouldComponentUpdate(prevProps, nextState) {}
//                      ~~~~~~~~~ Error: should be `nextProps`.
}
```

**Comment by ljharb** [comment]
Would it allow inline destucturing?

**Comment by golopot** [comment]
I don't see a reason to forbid destructuring.
